### DISCUSSION TITLE
"No such image" when running in docker container on Unrair

### DISCUSSION THREAD

**Discussion by MomoMiles** [nicolargo/glances#2258] [General]
**Describe the bug**
I am running into an issue where everytime I setup a fresh docker container with glances on unraid, it gives me a python error 

**To Reproduce**
1. Start a docker container using the template on the Community Apps page

**Expected behavior**
Launches properly

**Desktop (please complete the following information):**
 - Unraid v6.11.5

**Additional context**

```Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/docker/api/client.py", line 268, in _raise_for_status
    response.raise_for_status()
  File "/usr/lib/python3.10/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: http+docker://localhost/v1.41/images/e5cb7b52ed5c591bb57f9a53c36e1f7e657fe1d5e87d2da0a74feb8229ac0090/json

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/usr/lib/python3.10/site-packages/glances/__main__.py", line 19, in <module>
    glances.main()
  File "/usr/lib/python3.10/site-packages/glances/__init__.py", line 179, in main
    start(config=core.get_config(), args=core.get_args())
  File "/usr/lib/python3.10/site-packages/glances/__init__.py", line 108, in start
    mode = GlancesMode(config=config, args=args)
  File "/usr/lib/python3.10/site-packages/glances/webserver.py", line 31, in __init__
    self.stats.update()
  File "/usr/lib/python3.10/site-packages/glances/stats.py", line 216, in update
    self._plugins[p].update()
  File "/usr/lib/python3.10/site-packages/glances/plugins/glances_plugin.py", line 1114, in wrapper
    ret = fct(self, *args, **kw)
  File "/usr/lib/python3.10/site-packages/glances/plugins/glances_plugin.py", line 1131, in wrapper
    ret = fct(*args, **kw)
  File "/usr/lib/python3.10/site-packages/glances/plugins/glances_docker.py", line 255, in update
    container_stats['Image'] = container.image.tags
  File "/usr/lib/python3.10/site-packages/docker/models/containers.py", line 40, in image
    return self.client.images.get(image_id.split(':')[1])
  File "/usr/lib/python3.10/site-packages/docker/models/images.py", line 335, in get
    return self.prepare_model(self.client.api.inspect_image(name))
  File "/usr/lib/python3.10/site-packages/docker/utils/decorators.py", line 19, in wrapped
    return f(self, resource_id, *args, **kwargs)
  File "/usr/lib/python3.10/site-packages/docker/api/image.py", line 251, in inspect_image
    return self._result(
  File "/usr/lib/python3.10/site-packages/docker/api/client.py", line 274, in _result
    self._raise_for_status(response)
  File "/usr/lib/python3.10/site-packages/docker/api/client.py", line 270, in _raise_for_status
    raise create_api_error_from_http_exception(e) from e
  File "/usr/lib/python3.10/site-packages/docker/errors.py", line 39, in create_api_error_from_http_exception
    raise cls(e, response=response, explanation=explanation) from e
docker.errors.ImageNotFound: 404 Client Error for http+docker://localhost/v1.41/images/e5cb7b52ed5c591bb57f9a53c36e1f7e657fe1d5e87d2da0a74feb8229ac0090/json: Not Found ("no such image: e5cb7b52ed5c591bb57f9a53c36e1f7e657fe1d5e87d2da0a74feb8229ac0090: No such image: sha256:e5cb7b52ed5c591bb57f9a53c36e1f7e657fe1d5e87d2da0a74feb8229ac0090")```

**Comment by nicolargo** [comment]
Hi @MomoMiles 

i did not know Unraid. Perhaps ask the question directly to the Unraid Glances plugin maintainer: https://forums.unraid.net/profile/70671-atribe/

**Comment by rgabrielson11** [comment]
same issue here, but think this is not unraid related.  If I try to create the docker direct from the command line using 

docker run -d --restart="always" -p 61208-61209:61208-61209 -e GLANCES_OPT="-w" -v /var/run/docker.sock:/var/run/docker.sock:ro --name "Glances"--pid host nicolargo/glances:latest

i get

docker: Error response from daemon: pull access denied for host, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.


I did try logging in with my docker credentials from the command line, didnt seem to resolve.  According to my research this is due to one of the repos being set as private.

**Comment by MomoMiles** [comment]
> Hi @MomoMiles
> 
> i did not know Unraid. Perhaps ask the question directly to the Unraid Glances plugin maintainer: https://forums.unraid.net/profile/70671-atribe/

@nicolargo its not doing anything fancy. This is the docker run command it uses: 

```
docker run
  -d
  --name='glances-1-1'
  --net='host'
  -e TZ="America/Los_Angeles"
  -e HOST_OS="Unraid"
  -e HOST_HOSTNAME="MomoNAS"
  -e HOST_CONTAINERNAME="glances-1-1"
  -e 'GLANCES_OPT'='-w'
  -l net.unraid.docker.managed=dockerman
  -l net.unraid.docker.webui='http://[IP]:[PORT:61208]/'
  -l net.unraid.docker.icon='https://github.com/nicolargo/glances/raw/master/glances/outputs/static/images/glances.png'
  -v '/mnt':'/mnt':'ro'
  -v '/var/run/docker.sock':'/var/run/docker.sock':'ro'
  --pid=host 'nicolargo/glances'
```

**Comment by nicolargo** [comment]
Related to https://github.com/nicolargo/glances/issues/2233

**Comment by MomoMiles** [comment]
> Related to #2233

@nicolargo I saw that. Issue is that I have reinstalled multiple times on host, bridge, and my custom network. Same error everytime.

**Comment by nicolargo** [comment]
What is the version of the Glances image used ?

**Comment by MomoMiles** [comment]
I have tried Latest and Latest-Alpine I believe

**Comment by talondnb** [comment]
Unsure if related, but I’ve been getting the following from my watchtower notification for the past 6hrs:

Could not do a head request for "nicolargo/glances:latest-full", falling back to regular pull.
Reason: registry responded to head request with "404 Not Found", auth: "not present"

**Comment by tdharris** [comment]
I too get the same watchtower notification on multiple servers. Only started happening a day ago. It still seems to be able to later pull and recreate the container after it "falls back to regular pull" I assume.. curious though on the initially failed head request only happening recently.

**Comment by joeladria** [comment]
Same issue here... following...

**Comment by talondnb** [comment]
FYI, I've disabled the LABEL env to stop the monitoring on the glances container for the moment, this will stop the alerts at least.

**Comment by BadCo-NZ** [comment]
> Unsure if related, but I’ve been getting the following from my watchtower notification for the past 6hrs:
> 
> Could not do a head request for "nicolargo/glances:latest-full", falling back to regular pull. Reason: registry responded to head request with "404 Not Found", auth: "not present"

Same error for me from Watchtower.

Edit: Just to add, this is with Docker on a Ubuntu VM.

**Comment by nielsfranke** [comment]
> Unsure if related, but I’ve been getting the following from my watchtower notification for the past 6hrs:
> 
> Could not do a head request for "nicolargo/glances:latest-full", falling back to regular pull. Reason: registry responded to head request with "404 Not Found", auth: "not present"

Same here.

**Comment by psycik** [comment]
Same issue here.

`docker.errors.ImageNotFound: 404 Client Error for http+docker://localhost/v1.41/images/e95e16554ea2a72a7a2df99a68ccca4b8a5f5224331999d4edb11fe4cc42ca6a/json: Not Found ("no such image: e95e16554ea2a72a7a2df99a68ccca4b8a5f5224331999d4edb11fe4cc42ca6a: No such image: sha256:e95e16554ea2a72a7a2df99a68ccca4b8a5f5224331999d4edb11fe4cc42ca6a")`

**Comment by tdse13** [comment]
Same here:  level=warning msg="Reason: registry responded to head request with \"404 Not Found\", auth: \"not present\"" container=/glances image="nicolargo/glances:latest-full"

**Comment by nicolargo** [comment]
Sorry to hear that but it look like an Unrair issue.

The Glances docker images (latest-full) is available on Docker Hub:

![image](https://user-images.githubusercontent.com/776747/213858979-542166d1-974f-4816-a5c9-35e7e3fe4639.png)

**Comment by nielsfranke** [comment]
I am not using unraid and I get this error in watchtower.

This seems to affect other docker images as well: https://github.com/containrrr/watchtower/issues/1528

**Comment by talondnb** [comment]
Same here, no unraid here, purely docker with a watchtower monitoring the containers.

**Comment by kalyway** [comment]
If you all try it with this tag then it should work: `:alpine-3.2.6.4-full`

Found this fix on the Unraid forums. Not sure why anything else doesn't work.

**Comment by rgabrielson11** [comment]
Confirmed working with new tag. Thanks

**Comment by ghost** [comment]
Just chiming in. Watchtower user without unraid use here: same error.

Other containers seem to have updated to avoid the cause of the issue. Not sure this applies here but for the sake of spreading the information:

- [Koenkk/zigbee2mqtt](https://github.com/Koenkk/zigbee2mqtt/issues/16201?utm_source=pocket_saves#issuecomment-1399460127)
- [mention on containrrr/watchtower](https://github.com/containrrr/watchtower/issues/1528?utm_source=pocket_saves#issuecomment-1401320741)

**Comment by dsfsdbvxc** [comment]
Was having the same issue with updating on Unraid. Theres fix available that worked for me, details in the forum thread below:
[https://forums.unraid.net/topic/134016-fix-to-a-small-issue-with-updating-oci-docker-images/](url)

Some useful info related to the fix this Reddit post too:
[https://www.reddit.com/r/unRAID/comments/10pq4jf/dockers_not_updating_properly_patch_available/](url)
